app/src/main/java/com/scouts/kitchenplaner/ui/view/createrecipe/IngredientInput.kt

```diff
@@ -0,0 +1,200 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023-2024 Arne Kuchenbecker, Antonia Heiming
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+package com.scouts.kitchenplaner.ui.view.createrecipe
+
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.scouts.kitchenplaner.model.entities.Ingredient
+import com.scouts.kitchenplaner.ui.Headline
+import com.scouts.kitchenplaner.ui.view.NumberFieldType
+import com.scouts.kitchenplaner.ui.view.OutlinedNumberField
+
+@Composable
+fun IngredientsInput(
+    modifier: Modifier = Modifier,
+    ingredientGroups: Map<String, List<Ingredient>>,
+    onGroupAdd: (String) -> Unit,
+    onIngredientAdd: (String, Ingredient) -> Unit,
+    onIngredientDelete: (String, Ingredient) -> Unit,
+    onDeleteIngredientGroup: (String) -> Unit
+) {
+    println("Displaying ingredients: ${ingredientGroups.size}")
+    var addIngredientToGroup by remember { mutableStateOf("") }
+    var newGroupName by remember { mutableStateOf("") }
+    Column(
+        modifier = modifier
```

Comment:
Can you make a component for the box and the header? 
I think we can also reuse it for the create project screen and I really like the combination incl. header.